src/server/start.ts

```diff
@@ -31,9 +31,30 @@ import { PACKAGE } from '../shared/config'
 import { Server } from '@hapi/hapi'
 import { ServerApp } from './create'
 
-export default async function start(server: Server): Promise<Server> {
+let serverInstance: Server | null
+
+export async function start(server: Server): Promise<Server> {
   await server.start()
   const serverApp = server.settings.app as ServerApp
   logger.info(`Service '${PACKAGE.name}' is running '${serverApp.api} API' @ ${server.info.uri}`)
+  serverInstance = server
+  return server
+}
+
+export async function stop(server: Server): Promise<Server> {
+  await server.stop()
+  serverInstance = null
   return server
 }
+
+export async function restart(server: Server): Promise<void | Server> {
```

Comment:
I think it would be cleaner to have a factory class rather than separate methods in a module.
So instead of
```ts
import someModule from './server';
someModule.start(); // <--- it's unclear which module we access
```
we can have:
```ts
import { Server } from './server';
const server = Server::start(); // it's clear that we access Server class
```

Reply:
I'd rather just keep it functional for now vs extending off of `hapi.server`. I found it not that inituitive as well when messing about but lets leave that for a future refactor